Lombiq.Tests.UI/SecurityScanning/ZapManager.cs

```diff
@@ -0,0 +1,244 @@
+
+        await EnsureInitializedAsync();
+
+        if (string.IsNullOrEmpty(automationFrameworkYamlPath))
+        {
+            automationFrameworkYamlPath = AutomationFrameworkPlanPaths.BaselinePlanPath;
+        }
+
+        var mountedDirectoryPath = DirectoryPaths.GetTempSubDirectoryPath(context.Id, "Zap");
+        var reportsDirectoryPath = Path.Combine(mountedDirectoryPath, _zapReportsDirectoryName);
+        Directory.CreateDirectory(reportsDirectoryPath);
+
+        // Giving write permission to all users to the reports folder. This is to avoid issues under GitHub-hosted
+        // runners in GitHub Actions (BuildJet ones work without this too) at ZAP not being able to create the report.
+        // Pre-creating the report's folder would just prompt ZAP to try another folder name suffixed with "2".
+        if (GitHubHelper.IsGitHubEnvironment)
+        {
+            await new CliProgram("chmod").ExecuteAsync(_cancellationTokenSource.Token, "a+w", reportsDirectoryPath);
+        }
+
+        var yamlFileName = Path.GetFileName(automationFrameworkYamlPath);
+        var yamlFileCopyPath = Path.Combine(mountedDirectoryPath, yamlFileName);
+
+        File.Copy(automationFrameworkYamlPath, yamlFileCopyPath, overwrite: true);
+
+        await PreparePlanAsync(yamlFileCopyPath, modifyPlan);
+
+        // Explanation on the CLI arguments used below:
+        // - --add-host and --network host: Lets us connect to the host OS's localhost, where the OC app runs, with
+        //   https://localhost. See https://stackoverflow.com/a/24326540/220230. --network host serves the same, but
+        //   only works under Linux. See https://docs.docker.com/engine/reference/commandline/run/#network and
+        //   https://docs.docker.com/network/drivers/host/.
+        // - --volume: Mounts the given host folder as a volume under the given container path. This is to pass files
+        //   back and forth between the host and the container.
+        // - --tty: Allocates a pseudo-teletypewriter, i.e. redirects the output of ZAP to the CLI's output.
+        // - zap.sh: The entry point of ZAP. Everything that comes after this is executed in the container.
+
+        // Also see https://www.zaproxy.org/docs/docker/about/#automation-framework.
+
+        // Running a ZAP desktop in the browser with Webswing with the same config under Windows: #spell-check-ignore-line
+#pragma warning disable S103 // Lines should not be too long
+        // docker run --add-host localhost:host-gateway -u zap -p 8080:8080 -p 8090:8090 -i softwaresecurityproject/zap-stable zap-webswing.sh  #spell-check-ignore-line
+#pragma warning restore S103 // Lines should not be too long
+
+        var cliParameters = new List<object> { "run" };
+
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+        {
+            cliParameters.Add("--network");
+            cliParameters.Add("host");
+        }
+        else
```

Comment:
Does this work with Mac too? I'd guess that it rather needs the same as Linux, so probably need to swap the control flow and check against Windows OS instead.

Reply:
It doesn, only under Linux, [see here](https://docs.docker.com/network/drivers/host/#:~:text=The%20host%20networking%20driver%20only%20works%20on%20Linux%20hosts%2C%20and%20is%20not%20supported%20on%20Docker%20Desktop%20for%20Mac%2C%20Docker%20Desktop%20for%20Windows%2C%20or%20Docker%20EE%20for%20Windows%20Server.). I don't see any platform restrictions for `--add-host` (perhaps it could be used under Linux too, but from what I've read, `--network host` seems like the recommended approach).

Reply:
Gotcha, thanks!